app/main.py

```diff
@@ -1,2 +1,114 @@
+from dataclasses import dataclass
+from typing import Any, Hashable, List, Optional, Iterator
+
+
+INITIAL_CAPACITY = 8
+LOAD_FACTOR = 0.75
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+    _hash: int
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table: List[Optional[Node]] = [None] * self.capacity
```

Comment:
Maybe it would be better that you hash_table contains by default Nodes, not a None with Nodes?